src/components/builder.cpp

```diff
@@ -550,10 +550,10 @@ string builder::get_label_text(const label_t& label) {
   size_t maxlen = label->m_maxlen;
 
   if (maxlen > 0 && string_util::char_len(text) > maxlen) {
-    if (label->m_ellipsis) {
-      text = string_util::utf8_truncate(std::move(text), maxlen - 3) + "...";
-    }
-    else {
+    if (!label->m_ellipsis.empty()) {
+      text = string_util::utf8_truncate(std::move(text), maxlen - string_util::char_len(label->m_ellipsis)) +
```

Comment:
This code also works if ellipsis is empty (we truncate 0 characters). We don't need the extra check

Reply:
Fixed by https://github.com/polybar/polybar/pull/1673/commits/d2a6acbdfafdb6472993573a921f04deb8f261c7